reset-pass/account-validator-tool/src/java/org/sakaiproject/accountvalidator/tool/controller/MainController.java

```diff
@@ -0,0 +1,1283 @@
+            String labelLink = "<a href="+transferMembershipsURL+">"+linkText+"</a>";
+            model.addAttribute("wait2", labelLink);
+            model.addAttribute("addDetailsSub", messageSource.getMessage("submit.new.account", null, userLocale));
+            model.addAttribute("submitBtn", messageSource.getMessage("submit.new.account", null, userLocale));
+            //we need to know which sites they're a member of:
+            Set<String> groups = authzGroupService.getAuthzGroupsIsAllowed(EntityReference.getIdFromRef(va.getUserId()), "site.visit", null);
+            Iterator<String> git = groups.iterator();
+            List existingSites = new ArrayList<>();
+            while (git.hasNext()) {
+
+                String groupRef = git.next();
+                String groupId = EntityReference.getIdFromRef(groupRef);
+                if (!existingSites.contains(groupId)) {
+                    log.debug("groupId is " + groupId);
+
+                    try {
+                        Site s = siteService.getSite(groupId);
+                        model.addAttribute("siteListItem", s.getTitle());
+
+                        existingSites.add(groupId);
+                    
+                    } catch (IdUnusedException e) {
+
+                        log.error(e.getMessage(), e);
+                    }
+                }
+            }
+
+            if (existingSites.size() >= 2)  {
+                model.addAttribute("welcome", messageSource.getMessage("validate.welcome.plural", new String[]{uiService}, userLocale));
+
+            } else {
+                model.addAttribute("welcome", messageSource.getMessage("validate.welcome.single", new String[]{uiService}, userLocale));
+            }
+        }
+
+        model.addAttribute("welcome2", messageSource.getMessage("validate.welcome2", new String[]{u.getDisplayId()}, userLocale));
+
+        // We would gladly accept an alternate method of doing this, however
+        // this seems to be the only way to pass the enabled, disabled value from the server into the JavaScript
+        boolean passwordPolicyEnabled = (userDirectoryService.getPasswordPolicy() != null);
+        String passPolicyEnabledJavaScript = "VALIDATOR.isPasswordPolicyEnabled = " + Boolean.toString(passwordPolicyEnabled) + ";";
+        model.addAttribute("passwordPolicyEnabled", passPolicyEnabledJavaScript);
+
+        log.debug("account status: " + va.getAccountStatus());
+        boolean renderPassBox = false;
+
+        if (ValidationAccount.ACCOUNT_STATUS_NEW == va.getAccountStatus())  {
+
+            log.debug("this is a new account render the second password box");
+            renderPassBox = true;            
+        }
+        model.addAttribute("renderPassBox", renderPassBox);
+
+        // If we have some terms, get the user to accept them.
+        if (!"".equals(serverConfigurationService.getString("account-validator.terms"))) {
```

Comment:
Same as above with this, use a variable and StringUtils, do not call serverConfigurationService.getString two times.